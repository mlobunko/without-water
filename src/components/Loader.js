import React from "react";

export const Loader = ({ className }) => (
  <div className={`loader ${className}`} />
);
