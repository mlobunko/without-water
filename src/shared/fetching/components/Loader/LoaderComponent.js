import React from 'react';

export const LoaderComponent = ({ className }) => (
  <div className={`loader ${className}`} />
);

export default LoaderComponent;
