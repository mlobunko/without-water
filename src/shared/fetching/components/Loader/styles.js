import styled, { css } from 'styled-components';
import imageLoader from '../../images/loader.svg';

export const StyledLoader = styled.div`
  ${props =>
    props.centerPage &&
    css`
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      height: 60vh;
    `};
`;

export const StyledImage = styled.div`
  background: url(${imageLoader}) no-repeat center center;
  background-size: contain;
  height: 3rem;
  width: 100%;
  margin-bottom: ${props => props.login && '1rem'};
`;
