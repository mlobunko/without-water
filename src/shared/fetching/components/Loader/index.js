import Loader from './LoaderComponent';

export default Loader;
