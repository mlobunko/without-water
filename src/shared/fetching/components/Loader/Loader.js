import React from 'react';
import PropTypes from 'prop-types';
import { StyledImage, StyledLoader } from './styles';

export const Loader = ({ centerPage, login }) => (
  <StyledLoader centerPage={centerPage}>
    <StyledImage login={login} />
  </StyledLoader>
);

export default Loader;

Loader.propTypes = {
  centerPage: PropTypes.bool,
  login: PropTypes.bool
};
