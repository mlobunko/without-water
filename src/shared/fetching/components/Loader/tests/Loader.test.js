import React from 'react';
import { shallow } from 'enzyme';

import { Loader } from '../Loader';

describe('Loader component', () => {
  it('renders properly default', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders properly with full props', () => {
    const props = {
      centerPage: true,
      login: true
    };
    const wrapper = shallow(<Loader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
