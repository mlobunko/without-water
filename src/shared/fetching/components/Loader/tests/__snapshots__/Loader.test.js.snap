// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders properly default 1`] = `
<StyledLoader>
  <StyledImage />
</StyledLoader>
`;

exports[`Loader component renders properly with full props 1`] = `
<StyledLoader
  centerPage={true}
>
  <StyledImage
    login={true}
  />
</StyledLoader>
`;
